specification/sql/resource-manager/Microsoft.Sql/preview/2024-05-01-preview/InstancePoolOperations.json

```diff
@@ -0,0 +1,272 @@
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools/{instancePoolName}/operations": {
+      "get": {
+        "tags": [
+          "InstancePoolOperations"
+        ],
+        "description": "Gets a list of operations performed on the instance pool.",
+        "operationId": "InstancePoolOperations_ListByInstancePool",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ResourceGroupParameter"
+          },
+          {
+            "name": "instancePoolName",
+            "in": "path",
+            "description": "The name of the managed instance pool.",
+            "required": true,
+            "type": "string",
+            "maxLength": 127,
+            "minLength": 1,
+            "pattern": "^[a-z0-9]+(-*[a-z0-9]+)*$"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request for getting instance pool operations has been executed successfully.",
+            "schema": {
+              "$ref": "#/definitions/InstancePoolOperationListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 ManagedInstanceStoppingOrStopped - Conflicting operation submitted while instance is in stopping/stopped state\n\n * 400 ManagedInstanceStarting - Conflicting operation submitted while instance is in starting state\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List the instance pool management operations with no results": {
+            "$ref": "./examples/ListInstancePoolOperationsEmpty.json"
+          },
+          "List the instance pool management operations with some results": {
+            "$ref": "./examples/ListInstancePoolOperations.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools/{instancePoolName}/operations/{operationId}": {
```

Comment:
>"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/instancePools/{instancePoolName}/operations/{operationId} [](http://example.com/codeflow?start=4&length=149)

what about the Create/Update/Delete of this operation? how are they orchestrated ?

Reply:
Instance pool operations cannot be create, updated and deleted via API. They are implicitly created by the control plan when crate/update/delete is executed on the instance pool it self.

This is why we have only list and get for instance pool operations. Same patter is used for SQL Managed Instance (and SQL DB, I believe).

Reply:
btw, this was already answered in the PR introducing this change.